### PR TITLE
Nav SDK v3 examples

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
This PR adds examples for various Nav SDK scenarios, like:
- Camera transition
- Buildings highlight
- Voice instructions
- Draw route line
- Copilot

### CODE REVIEW

**Review by DSamaryan — APPROVED** [review]

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Copilot is a library included in the Navigation SDK that processes full-trip-trace longitude and
+ * latitude data ("**Copilot**"). Copilot is turned off by default, and can be enabled by you at the
+ * application-developer level to improve feedback resolution. If you enable Copilot, your organization is responsible
+ * for obtaining and maintaining all necessary consents and permissions, including providing notice to and obtaining your end
+ * users' affirmative, expressed consent before any access or use of Copilot.
+ */
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CopilotViewModel : ViewModel() {
```
Comment (VysotskiVadim) [inline app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotViewModel.kt:33]: why do we need view model for copilot if none of our examples can handle configuration change?

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotActivity.kt [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                        null,
+                    )
+                )
+            )
+            Toast.makeText(this, "Search events pushed!", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun buildTestRoute() {
+        val origin = Point.fromLngLat(-77.0339911055176, 38.899920004207516)
```
Comment (VysotskiVadim) [inline app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotActivity.kt:232]: Route from hardcoded origin will most likely cause reroute without location simulation. Not sure if that's important in the context of this example

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/voice/PlayVoiceInstructionsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+                .build(),
+            mapAnimationOptions
+        )
+    }
+
+    private fun startNavigation(routes: List<NavigationRoute>) {
+        if (routes.isEmpty()) return
+
+        mapboxNavigation.setNavigationRoutes(routes)
+        startSimulation(routes.first().directionsRoute)
```
Comment (VysotskiVadim) [inline app-preview/src/main/java/com/mapbox/navigation/examples/preview/voice/PlayVoiceInstructionsActivity.kt:376]: is that needed? `ReplayProgressObserver` doesn't start simulation, does it?

Comment (VysotskiVadim) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/voice/PlayVoiceInstructionsActivity.kt:376]: BTW, maybe we should promote `ReplayRouteSession` in our examples?

Comment (VysotskiVadim) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/voice/PlayVoiceInstructionsActivity.kt:376]: I found out that in other examples we use it in the same way, so never mind

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotViewModel.kt (hunk 1 above)
Comment (DzmitryFomchyn) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotViewModel.kt:33]: This is just a revert of recently removed copilot example https://github.com/mapbox/mapbox-navigation-android-examples/pull/198/files#diff-76da0a5a89b901d013d4e8f7a823640e6259d13b4bee1d92083bee18f6a48a93

It doesn't really matter, copilot cab be started/stopped in activity callbacks directly.

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotActivity.kt (hunk 2 above)
Comment (DzmitryFomchyn) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotActivity.kt:232]: Correct, it's not important in this example. CopilotActivity just shows how to integrate copilot and send events.

**Comment by DzmitryFomchyn** [comment]
Closing in favor of https://github.com/mapbox/mapbox-navigation-android-examples/pull/201

**Pull request closed without merging**